packages/protocol/engine/consensus/conflictresolver/conflictresolver.go

```diff
@@ -69,53 +75,55 @@ func (o *ConflictResolver) ConflictLiked(conflictID utxo.TransactionID) (conflic
 }
 
 // conflictPreferred returns whether the conflict is the winner across its conflict sets.
-func (o *ConflictResolver) conflictPreferred(conflictID utxo.TransactionID, likeWalker *walker.Walker[utxo.TransactionID]) (preferred bool) {
+func (o *ConflictResolver) conflictPreferred(conflict *conflictdag.Conflict[utxo.TransactionID, utxo.OutputID], likeWalker *walker.Walker[*conflictdag.Conflict[utxo.TransactionID, utxo.OutputID]]) (preferred bool) {
 	preferred = true
-	if conflictID == utxo.EmptyTransactionID {
+	if conflict.ID() == utxo.EmptyTransactionID {
 		return
 	}
+	// TODO: this should never happen?
 
-	o.conflictDAG.Storage.CachedConflict(conflictID).Consume(func(conflict *conflictdag.Conflict[utxo.TransactionID, utxo.OutputID]) {
-		switch conflict.ConfirmationState() {
-		case confirmation.Rejected:
-			preferred = false
-			return
-		case confirmation.Accepted:
-		case confirmation.Confirmed:
-			return
-		}
+	switch conflict.ConfirmationState() {
+	case confirmation.Rejected:
+		preferred = false
+		return
+	case confirmation.Accepted:
+	case confirmation.Confirmed:
+		return
+	}
 
-		if preferred = !o.dislikedConnectedConflictingConflicts(conflictID).Has(conflictID); preferred {
-			for it := conflict.Parents().Iterator(); it.HasNext(); {
-				likeWalker.Push(it.Next())
+	if preferred = !o.dislikedConnectedConflictingConflicts(conflict).Has(conflict.ID()); preferred {
+		for it := conflict.Parents().Iterator(); it.HasNext(); {
+			parentConflict, exists := o.conflictDAG.Conflict(it.Next())
+			if !exists {
```

Comment:
why not less code?

`if exists {
				likeWalker.Push(parentConflict)
			}
			`